api/config.go

```diff
@@ -147,5 +157,50 @@ func LoadDefaultConfig(args []string) (config Config, err error) {
 	if err = flagSet.Parse(args[1:]); err != nil {
 		return config, err
 	}
+
+	if config.Flag3DTConfig != "" {
+		return readConfigFile(config.Flag3DTConfig, config)
+	}
 	return config, nil
 }
+
+func readConfigFile(configPath string, originalConfig Config) (Config, error) {
+	// load config
+	configContent, err := ioutil.ReadFile(configPath)
+	if err != nil {
+		return originalConfig, err
+	}
+
+	var configJSON map[string]interface{}
```

Comment:
You can unmarshal directly to your `Config{}` struct, the JSON tags enable validation on the fly.

See: https://github.com/dcos/dcos-signal/blob/master/config/config.go#L96

You can actually do this over and over again on the same struct pointer if you have more than one config file.